File: src/TheWorld/wwwroot/javascript/tripsController.js
(function () {
    "use strict";

    angular.module("app-trips")
           .controller("tripsController", tripsController);

    function tripsController($http) {
        var viewModel = this;
        viewModel.trips = [];
        viewModel.newTrip = {};
        viewModel.errorMessage = "";
        viewModel.isBusy = true;

        $http.get("/api/trips")
            .then(function (response) {
                angular.copy(response.data, viewModel.trips);
            }, function (error) {
                viewModel.errorMessage = "Unable to load data: " + error;
            })
            .finally(function () {
                viewModel.isBusy = false;
            });

        viewModel.addTrip = function () {
            viewModel.isBusy = true;
            viewModel.errorMessage = "";

            $http.post("/api/trips", viewModel.newTrip)
                .then(function (response) {
                    viewModel.trips.push(response.data);
                    viewModel.newTrip = {};
                }, function () {
                    viewModel.errorMessage = "Unable to save trip";
                })
                .finally(function () {
                    viewModel.isBusy = false;
                });
        };
    }
})();

File: src/TheWorld/Controllers/Api/TripsController.cs
using AutoMapper;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using TheWorld.Entities;
using TheWorld.ViewModels;

namespace TheWorld.Controllers.Api
{
    [Authorize]
    [Route("api/trips")]
    public class TripsController : Controller
    {
        private ILogger<TripsController> logger;
        private IWorldRepository repository;

        public TripsController(IWorldRepository repository, ILogger<TripsController> logger)
        {
            this.repository = repository;
            this.logger = logger;
        }

        #region Methods
        [HttpGet("")]
        public IActionResult Get()
        {
            try
            {
                var results = this.repository.GetTripsByUserName(User.Identity.Name);
                return Ok(Mapper.Map<IEnumerable<TripViewModel>>(results));
            }
            catch (Exception ex)
            {
                this.logger.LogError($"An Exception occured getting trips: {ex}");
                return BadRequest("An Error Occured.");
            }

        }

        [HttpPost("")]
        public async Task<IActionResult> Post([FromBody] TripViewModel theTrip)
        {
            if (ModelState.IsValid)
            {
                var newTrip = Mapper.Map<Trip>(theTrip);
                newTrip.UserName = User.Identity.Name;
                repository.AddTrip(newTrip);
                if (await this.repository.SaveChangesAsync())
                {
                    return Created($"api/trips/{theTrip.Name}", Mapper.Map<TripViewModel>(newTrip));
                }
            }
            return BadRequest("An error occured while saving the trip");
        }
        #endregion
    }
}
